bolt/lib/Profile/DataAggregator.cpp

```diff
@@ -804,9 +804,13 @@ bool DataAggregator::doBranch(uint64_t From, uint64_t To, uint64_t Count,
   };
 
   BinaryFunction *FromFunc = handleAddress(From, /*IsFrom=*/true);
-  // Ignore returns.
-  if (IsReturn)
-    return true;
+  // Record returns as call->call continuation fall-through.
+  if (IsReturn) {
```

Comment:
Does the fix affect regular (non pre-aggregated) profiles as well?

Reply:
Yes, it actually does and it will result in double-counting in perf data. 

I'm thinking whether I should restrict new handling to pre-aggregated profile only, or to drop the heuristic used for perf data. I'll do perf experiments and look at resulting profile continuity to decide.